snappi_ixnetwork/capture.py

```diff
@@ -155,57 +179,395 @@ def _config_missing_pallete(self, cap_filter, pallette, trigger, filter):
                     else:
-                        filter[
-                            "captureFilterPattern"
-                        ] = new_pattern.filter_pattern
-                    trigger["triggerFilterPattern"] = filter[
+                        filter["captureFilterPattern"] = (
+                            new_pattern.filter_pattern
+                        )
+                    trigger["captureTriggerPattern"] = filter[
                         "captureFilterPattern"
                     ]
+                    if len(expressionString) == 0:
+                        expressionString = negate + "P1"
+                    else:
+                        expressionString = (
+                            expressionString + " and " + negate + "P1"
+                        )
 
-    def _config_custom_pallete(self, cap_filter, pallette, trigger, filter):
+        return expressionString
+
+    def _config_custom_pallete(
+        self, cap_filter, pallette, trigger, filter, expressionString
+    ):
         if cap_filter.value is not None:
             pallette["pattern1"] = cap_filter.value
             if cap_filter.mask is not None:
                 pallette["patternMask1"] = cap_filter.mask
             if cap_filter.offset is not None:
                 pallette["patternOffset1"] = cap_filter.offset
+            negate = ""
             if cap_filter.negate is not None and cap_filter.negate is True:
                 filter["captureFilterPattern"] = "notPattern1"
+                negate = "!"
             else:
                 filter["captureFilterPattern"] = "pattern1"
-            trigger["triggerFilterPattern"] = filter["captureFilterPattern"]
+            trigger["captureTriggerPattern"] = filter["captureFilterPattern"]
+            if len(expressionString) == 0:
+                expressionString = negate + "P1"
+            else:
+                expressionString = expressionString + " and " + negate + "P1"
 
-    def _config_ethernet_pallette(self, cap_filter, pallette, trigger, filter):
-        if cap_filter.src.value is not None:
-            pallette["SA1"] = cap_filter.src.value
-            if cap_filter.src.mask is not None:
-                pallette["SAMask1"] = cap_filter.src.mask
-            if (
-                cap_filter.src.negate is not None
-                and cap_filter.src.negate is True
-            ):
+        return expressionString
+
+    def _config_ethernet_pallette(
+        self, cap_filter, pallette, trigger, filter, expressionString
+    ):
+        if not cap_filter.src.value == "00":
+            expressionString = self._config_source_address(
+                cap_filter.src, pallette, trigger, filter, expressionString
+            )
+        if not cap_filter.dst.value == "00":
+            expressionString = self._config_destination_address(
+                cap_filter.dst, pallette, trigger, filter, expressionString
+            )
+        if not cap_filter.ether_type.value == "00":
```

Comment:
This is not a must change. 
calling multiple `_config_common_filter_pallete` is little annoying. is it not possible to maintain a class has all those property in a dics. Then loop through and call  `_config_common_filter_pallete` once.